Replace deprecated Buffer usage

oliversalzburg opened the pull request:
[The `Buffer` constructor is deprecated](https://nodejs.org/en/docs/guides/buffer-constructor-deprecation/). This change replaces all usages with the safe alternatives.

oliversalzburg commented:
@brianc How do you want to see this resolved? Drop NodeJS <5.9 support? Include a polyfill to support old versions?

brianc commented:
I think dropping support for`<4.x` is totally fine, I'll just bump a major semver.  I believe `Buffer.alloc` is available in node 4 lts, right?

oliversalzburg commented:
@brianc This migration guide says that 4.5 should be good. So, it should be fine.

I just realized that the test failures on Node 4/5 are due to the mocha version in use depending on `debug@*`, which pulls in the absolute latest version, which is not compatible with Node 4/5.

I adjusted the targets in the Travis config and replaced the mocha dependency with a newer version that doesn't have the same dependency.

Let me know if I need to change anything.

kibertoad commented:
Is there anything left before this can be merged?..

oliversalzburg commented:
@kibertoad Not from my side. Unless changes are requested, I feel like this is ready to be merged.

vitaly-t commented:
Let's keep min Node.js version at 4.5, it all works fine within `pg-promise`.

kibertoad commented:
@brianc Any updates?..

brianc commented:
looks good - will release a new version today.

oliversalzburg commented:
@vitaly-t Could you bump `pg` in `pg-promise` now?

vitaly-t commented:
@oliversalzburg done ;)
